src/test/common/extensionMessaging.test.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {HostPlatform} from "../../common/hostPlatform";
+
+import {
+    ExtensionMessage,
+    ExtensionMessageSender
+} from "../../common/extensionMessaging";
+
+import * as assert from "assert";
+import * as net from "net";
+import * as Q from "q";
+
+let mockServer: net.Server;
+
+suite("extensionMessaging", function() {
+    suite("commonContext", function() {
+        teardown(function() {
+            if (mockServer) {
+                mockServer.close();
```

Comment:
Close takes a callback. We should turn this into a promise, and wait until the server is fully closed.